### PR TITLE
signupConfirmed refactor: photoPostConfig.startPhotoPost + textPostConfig.askText

### PR DESCRIPTION
**Pull request by aaronschachter**
## Update
4/16 - I'm going to hold off on renaming and migrating any fields in Contentful, as we'll want to start using code for content migrations #1037. 

When I revisited the `menu-doing` and `menu-completed` middleware to rename the replies to `botSignupConfirmed` and `photoPostCompleted`, it got me thinking that it felt pretty confusing that a `botSignupConfirmed` template could have two different functions, it asks for START in a photoPostConfig and asks for reportback text in a textPostConfig.  It also seemed logical to combine the `menu-doing` and `menu-completed` middleware into a single `draft-not-found` to DRY.  Seeing the logic for both doing/completed in one place made it seem like renaming `botSignupConfirmed` as `startPhotoPost`  would help make things more readable.

For now, this PR uses the existing field names to avoid making too many potentially breaking changes at once. I'll be opening up a new Conversations PR to parse the new template names as campaign activity messages.

#### What's this PR do?
Adds new templates for a `photoPostConfig` content type per the Gambini spec in #1033, adding new fields in Contentful:
*  `botSignupConfirmedMessage`
*  `webSignupConfirmedMessage`
*  `signupConfirmedAutoReply`
*  `photoPostCompleted`
* `photoPostCompletedAutoReply`

#### How should this be reviewed?
Once the fields have been added and their values have been backfilled, verify the copy is identical in a `GET /campaigns/:id` response for each pair of templates, listed as new template + to-be-deprecated template.

* botSignupConfirmed + gambitSignupMenu
* webSignupConfirmed + externalSignupMenu
* signupConfirmedAutoReply + invalidSignupMenuCommand
* photoPostCompleted + completedMenu
* photoPostCompletedAutoReply + invalidCompletedMenuCommand

#### Any background context you want to provide?
I'll be checking off each field in this PR as I add them into Contentful, and will go through each `photoPostConfig` entry in production to copy values over.  Contentful migration CLI could be nice for this... but there's only 7 entries so it doesn't feel like a huge time sink 🚰 

#### Relevant tickets
* #1033 
* Completes last task in #1021 

#### Checklist
- [x] Documentation added for new features/changed endpoints.
- [x] Tested on staging.

### CODE REVIEW

**Comment by aaronschachter** [comment]
I've updated this PR description and tested on staging, this is ready for review along with proposed field names in the bottom section of the botConfig spec.

File: lib/replies.js [hunk 1]
```diff
@@ -42,24 +42,20 @@ module.exports.invalidWhyParticipated = function invalidWhyParticipated(req, res
...
+  return sendResponse('completedPhotoPostAutoReply', req, res);
 };
 
-module.exports.menuSignedUp = function menuSignedUp(req, res) {
-  return sendResponse('gambitSignupMenu', req, res);
+module.exports.startPhotoPost = function startPhotoPost(req, res) {
+  return sendResponse('startPhotoPost', req, res);
 };
 
-module.exports.externalSignupMenu = function externalSignupMenu(req, res) {
```
Comment (aaronschachter) [inline lib/replies.js]: This is a pre-TGM relic, a `POST /campaignActivity` request would never return an `externalSignupMenu` reply template anymore, confirmation messages for web signups are now handled via Gambit Conversations `POST /v2/messages?origin=signup`.

File: lib/replies.js [hunk 2]
```diff
@@ -70,22 +66,17 @@ module.exports.noPhotoSent = function noPhotoSent(req, res) {
   return sendResponse('invalidPhoto', req, res);
 };
 
-module.exports.campaignClosed = function campaignClosed(req, res) {
-  return sendResponse('campaignClosed', req, res);
-};
-
-module.exports.memberSupport = function memberSupport(req, res) {
-  return sendResponse('memberSupport', req, res);
-};
```
Comment (aaronschachter) [inline lib/replies.js]: These are more pre-TGM relics, a `POST /campaignActivity` request does not check campaign status nor for support requested commands.

File: app/routes/campaignActivity.js [hunk 3]
```diff
@@ -12,17 +12,18 @@ const mapRequestParamsMiddleware = require('../../lib/middleware/campaignActivit
...
 const createNewSignupIfNotFoundMiddleware = require('../../lib/middleware/campaignActivity/signup-create');
 const validateRequestMiddleware = require('../../lib/middleware/campaignActivity/validate');
+
 // Middleware for Text Posts.
 const createTextPostMiddleware = require('../../lib/middleware/campaignActivity/text/post-create');
+
 // Middleware for Photo Posts.
-const doingMenuMiddleware = require('../../lib/middleware/campaignActivity/photo/menu-doing');
 const createDraftSubmissionMiddleware = require('../../lib/middleware/campaignActivity/photo/draft-create');
+const draftSubmissionNotFoundMiddleware = require('../../lib/middleware/campaignActivity/photo/draft-not-found');
```
Comment (rapala61) [inline app/routes/campaignActivity.js]: 👏

File: config/lib/helpers/botConfig.js [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
+const startCommand = '{{cmd_reportback}}';
+const startPhotoPostText = `Thanks for signing up for {{title}}! Text ${startCommand} to submit a post.`;
+const completeAnotherPhotoPostText = `To submit another post for {{title}}, text ${startCommand}`;
+const photoPostDefaultText = {
+  askWhyParticipated: 'Last question: Why was participating in {{title}} important to you? (No need to write an essay, one sentence is good).',
+  completedPhotoPost: `Great! We've got you down for {{quantity}}. ${completeAnotherPhotoPostText}`,
+  completedPhotoPostAutoReply: `${defaultText.invalidInput}\n\n${completeAnotherPhotoPostText}`,
+  startPhotoPost: startPhotoPostText,
+  startPhotoPostAutoReply: `${defaultText.invalidInput}\n\nText ${startCommand} when you're ready to submit a post for {{title}}.`,
+  webStartPhotoPost: `Hi it's Freddie from DoSomething!${startPhotoPostText}`,
```
Comment (rapala61) [inline config/lib/helpers/botConfig.js]: Maybe just add a space before `${startPhotoPostText}`

File: lib/replies.js (hunk 2 above)
Comment (rapala61) [reply lib/replies.js]: Nice pruning! ✂️

**Review by rapala61 — APPROVED** [review]
Just a super minor thing. Really great work with the latest template naming and DRYing up code. 🥇

**Comment by rapala61** [comment]
screen shot 2018-04-17 at 2 31 34 pm

[#2meirl4meirl](https://www.reddit.com/r/2meirl4meirl/)

**Pull request merged**